src/dist/dist.rs

```diff
@@ -657,6 +666,8 @@ fn update_from_dist_<'a>(
                     if first_err.is_none() {
                         first_err = Some(e);
                     }
+                    // We decrement the backtrack count only on unavailable component errors
```

Comment:
I think it'd be good to have this comment explain _why_ we do that. In particular

> so that the limit only applies to nightlies that were indeed available, and ignores missing ones.